src/ds/logger.h

```diff
@@ -117,17 +132,64 @@ namespace logger
     }
   };
 
+  class formatter
+  {
+  public:
+    static void format(
+      const std::string& file_name,
+      int line_number,
+      const std::string& ll_str,
+      const std::string& msg,
+      const std::string& host_ts,
+      const std::optional<std::string>& enclave_ts = std::nullopt)
+    {
+      nlohmann::json j;
+      j["file"] = file_name;
+      j["number"] = line_number;
+      j["level"] = ll_str;
+      j["msg"] = msg;
+      j["h_ts"] = host_ts;
+      if (enclave_ts.has_value())
+      {
+        j["e_ts"] = enclave_ts.value();
+      }
+
+      dump(j.dump(), config::custom_path());
```

Comment:
Minor: `j.dump(2)` will pretty-print and indent, but if this is entirely for machine consumption then it doesn't matter.

Reply:
This is for machine consumption and it should be one log per line

Reply:
Humans are expected to read the string-formatted logs instead ;) Also this can always be piped into python -msjon.tool if needed.